### DISCUSSION TITLE
Why the change from black to ruff?

### DISCUSSION THREAD

**Discussion by peterhinch** [micropython/micropython#12914] [General]
Is there an issue with black or is it simply to perform linting in addition to formatting?

**Comment by jimmo** [comment]
No issue with black specifically. Like you say, seeing as we already had ruff linting, we now have one tool to do both (i.e. one fewer dependency). It's also quite a lot faster (which is minor but noticeable when you're doing whole-repo checking).

**Comment by andrewleech** [comment]
`ruff format` is generally compatible with black, but with a couple of documented differences which I feel ruff does better: https://github.com/astral-sh/ruff/blob/main/docs/formatter.md#intentional-deviations

https://astral.sh/blog/the-ruff-formatter

**Comment by sosi-deadeye** [comment]
The speed of ruff is ridiculous.

**Comment by andrewleech** [comment]
I was just wondering how much faster black would be if they ran it through something like cpython... Turns out they already do! (mypyc)
https://ichard26.github.io/blog/2022/05/compiling-black-with-mypyc-part-1/

TLDR it only yielded a 2x speed improvement over pure python.

**Comment by peterhinch** [comment]
Thanks all for clarifying.
